test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -976,6 +977,197 @@ public static CommandRunnerResult RunRestore(SimpleTestPathContext pathContext,
             return r;
         }
 
+        [Fact]
+        public async Task Restore_PackageNamespace_Succeed()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var net461 = NuGetFramework.Parse("net461");
+
+                var projectA = SimpleTestProjectContext.CreateLegacyPackageReference(
+                    "a",
+                    pathContext.SolutionRoot,
+                    net461);
+                var projectAPackages = Path.Combine(pathContext.SolutionRoot, "packages");
+
+                var externalRepositoryPath = Path.Combine(pathContext.SolutionRoot, "ExternalRepository");
+                Directory.CreateDirectory(externalRepositoryPath);
+
+                var contosoRepositoryPath = Path.Combine(pathContext.SolutionRoot, "ContosoRepository");
+                Directory.CreateDirectory(contosoRepositoryPath);
+
+                var configPath = Path.Combine(pathContext.WorkingDirectory, "nuget.config");
+                SettingsTestUtils.CreateConfigurationFile(configPath, $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""ExternalRepository"" value=""{externalRepositoryPath}"" />
+    <add key=""ContosoRepository"" value=""{contosoRepositoryPath}"" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key=""externalRepository"">
+            <namespace id=""External.*"" />
```

Comment:
I see tests for wildcards at the end of a package ID, but what about:

* wildcard at the beginning
* wildcard in the middle
* only a wild card
* multiple wildcards

(If you have tests somewhere already, great.  I didn't find them though.)

I think the spec should be explicit about where and how many wildcards are supported.  If some of the above scenarios are not supported, it's still worth testing them (as negative tests).

CC @nkolev92

Reply:
I believe here is issue for this https://github.com/NuGet/Home/issues/10783